components/scream/src/share/scream_logging.hpp

```diff
@@ -0,0 +1,271 @@
+
+#ifndef SCREAM_LOGGING_HPP
+#define SCREAM_LOGGING_HPP
+
+#include <cstdio>
+#include <cstdlib>
+#include <string>
+#include <vector>
+#include <stdexcept>
+#include "scream_config.h"
+
+#if LOG_STACK_TRACE
+#include <execinfo.h>
+#endif
+
+namespace scream {
+namespace util {
+struct Error : public std::runtime_error {
+  explicit Error(const std::string &s) : std::runtime_error(s) {}
+};
+}  // namespace util
+}  // namespace scream
+
+#include <assert.h>
+#include <iostream>
+#include <sstream>
+#include <ctime>
+
```

Comment:
I think a detailed comment should be added here. What exactly is this header providing? How should developers use this stuff?

Reply:
it is very simple,
#include "share/scream_logging.h"
....
LOG(INFO) << "anything you want add" << std::endl;
or 
LOG(WARNING)
LOG(FATAL)
for different level of log message

Reply:
we also can use the SCREAM_CHECK instead of exceptions, exceptions will force the code to exit if throw an error as we have in the scream_assert code.

Reply:
Where does the output go? Why would I use this instead of printf? How do I turn on/off various levels of logging? This really needs to be documented.